### PR TITLE
JUnit Upgrade

### PR DESCRIPTION
**Pull request by br648**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

This PR aims to update JUnit from version 4 to version 5. This includes updating all JUnit references (e.g. Test, BeforeClass, Before etc), updating deprecated classes (e.g. to the org.hamcrest package) and refactoring tests to use Wiremock without the deprecated @Rule annotation.

Fixes: ibi-group/datatools-server#362

### CODE REVIEW

File: src/test/java/com/conveyal/datatools/manager/controllers/api/UserControllerTest.java [hunk 1]
```diff
@@ -42,34 +40,32 @@
  * certain responses needed to verify functionality.
  */
 public class UserControllerTest extends UnitTest {
-    private String emailForExistingAccount = "test-existing-user@test.com";
-    private ObjectMapper mapper = new ObjectMapper();
+    private final String emailForExistingAccount = "test-existing-user@test.com";
+    private final ObjectMapper mapper = new ObjectMapper();
     private final String USER_SEARCH_API_PATH = USERS_API_PATH + "?sort=email%3A1&per_page=10&page=0&include_totals=false&search_engine=v3&q=email%3A";
-    /**
-     * This sets up a mock server that accepts requests and sends predefined responses to mock an Auth0 server.
```
Comment (landonreed) [inline src/test/java/com/conveyal/datatools/manager/controllers/api/UserControllerTest.java:49]: This comment (which I think is nice to have) was lost.

**Review by landonreed — APPROVED** [review]
LGTM, just the one comment about the wiremock stuff.

File: src/test/java/com/conveyal/datatools/manager/controllers/api/UserControllerTest.java (hunk 1 above)
Comment (br648) [reply src/test/java/com/conveyal/datatools/manager/controllers/api/UserControllerTest.java:49]: Updated.

**Review by binh-dam-ibigroup — APPROVED** [review]
Looks good!

**Review by evansiroky — APPROVED** [review]

**Pull request merged**